services/lsst-dev.rst

```diff
@@ -116,39 +116,41 @@ Select Appropriate Developer Tools
 
 The ``lsst-dev01`` system is configured with the latest CentOS 7.x as its operating system.
 This release of CentOS provides an old set of development tools, centered around version 4.8.5 of the `GNU Compiler Collection`_ (GCC).
-While this version of GCC does satisfy the `prerequisites for building the LSST stack`_, it is still rather out of date.
-However, this version of GCC is used for building the :ref:`shared stacks <lsst-dev-loadlsst>` available on the machine and must be used if building against those stacks.
-It is also used by the ``lsstsw`` account that builds releases.
+Updated toolchains are made available through the “Software Collection” system.
+The following Software Collections are currently available:
 
-If you prefer, you can enable the `Red Hat Developer Toolset`_ version 3 (``devtoolset-3``) which has been pre-installed.
-This provides an updated toolchain, including GCC 4.9.2.
-If you enable this, you must not use the shared stack; instead you must build your own version of the entire stack using this toolchain.
+================ ================================================
+Name             Description
+================ ================================================
+``devtoolset-3`` Updated compiler toolchain providing GCC 4.9.2.
+``devtoolset-4`` Updated compiler toolchain providing GCC 5.3.1.
+``devtoolset-6`` Updated compiler toolchain providing GCC 6.3.1.
+``devtoolset-7`` Updated compiler toolchain providing GCC 7.1.1.
+``git19``        The `Git`_ version control system version 1.9.4.
+================ ================================================
 
-Enable and test ``devtoolset-3`` using the ``scl`` command as follows (replacing ``bash`` with your shell of choice if necessary):
+To enable a particular Software Collection use the ``scl`` command. For example:
 
 .. prompt:: bash $ auto
 
-   $ scl enable devtoolset-3 bash
+   $ scl enable devtoolset-6 bash
    $ gcc --version
-   gcc (GCC) 4.9.2 20150212 (Red Hat 4.9.2-6)
-   Copyright (C) 2014 Free Software Foundation, Inc.
+   gcc (GCC) 6.3.1 20170216 (Red Hat 6.3.1-3)
+   Copyright (C) 2016 Free Software Foundation, Inc.
    This is free software; see the source for copying conditions.  There is NO
    warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
-The Developer Toolset includes version 1.9.3 of the `Git`_ version control system.
-If you prefer the (slightly) more recent version 1.9.4, you may also wish to enable the ``git19`` package.
-This may be done at the same time as enabling ``devtoolset-3``.
+.. warning::
 
-.. prompt:: bash
-
-   scl enable devtoolset-3 git19 bash
+   Code compiled by different versions of GCC may not be compatible: it is generally better to stick to a particular toolchain for a given project.
+   In particular, if you are using a :ref:`shared stack <lsst-dev-loadlsst>` you *must* use the matching toolchain.
```

Comment:
Since all our stacks currently use the same toolchain, and this guide is specifically for DM developers, I wonder if we shouldn't be even more explicit here about what people should do:

```If you are using a shard stack<link>, put the following into your .profile: ```

or something like that. Plus, then we can be sure people get the correct `if hostname == blah` or
```bash
if [[ $- != *i* ]] ; then
    # Shell is non-interactive.
    return
fi
```
stuff so as to not get errors when running batch jobs (because `scl` is not available on the cluster).

Reply:
I'll try to rework the wording to address this concern to some extent. I'm worried that it's possible in general, though. We explicitly support multiple shells (we provide four different `loadLSST` scripts), and people have their environments configured in different ways: trying to explicitly document what to do in every possible circumstance is hard (and honestly, I've no idea what the correct incantations are in ksh, nor do I think it's a good use of time to find out).

Anyway, I'll see if I can come up with some sort of wording which is a halfway house.

Reply:
I guess I'd say "provide a good solution for bash" (almost certainly the most common shell) to prevent most of us from having to come up with our own. There were some possible solutions on slack#dm-infrastructure to choose from, but most DM developers are likely not shell wizards (see my forkbombing), so a standard shell setup would be widely used.

Reply:
Oh, and my comment about this also extend to getting a modern version of `git` setup as well